### ISSUE TITLE
conj:emb; converter 2.7

### ISSUE THREAD

**Issue by ALeczkowski** [surfacesyntacticud/tools#13]
Hello, 

I am struggling to understand the usage of the label conj:emb in the file UD_to_SUD.grs.

There is no such label (conj:emb) in UD treebanks. However, before introducing it in the rule `add_conj_emb`, you use it in several rules before that e.g. in the `rule change_ext` or in the `rule minimize_direct`. 

I was wandering whether there is a problem with my understanding of the code or it is a mistake/an artefact of previous versions of the code.

**Comment by perrier54** [comment]
In SUD, the conjuncts of a coordination are organized in a chain, unlike in UD where they are organized in a bouquet. This is why we need the extension `emb' to express the embedded coordinations in SUD.
For example:
``` Les stages sont supers, les enfants font des progrès en anglais et se sont épanouis. (The courses are great, the children are making progress in English and have flourished.) ```

In UD, we have : `super -[conj]-> font; font -[conj]-> épanouis`
If the second coordination was not embedded, the annotation would be: `super -[conj]-> font; super -[conj]-> épanouis`
There is no need for an extension to express embedded coordination.

In SUD, we have: `super -[conj]-> font; font -[conj:emb]-> épanouis`
If the second coordination was not embedded, the annotation would be: `super -[conj]-> font;  font -[conj]-> épanouis`
Extension is necessary to express the embedded coordination.

The rule `change_ext` changes UD extensions to SUD deep extensions, which usually express semantic properties. Since   the extension `emb` does not express any semantic properties, it remains unchanged.

Now , the presence of the extension `emb` in the rule `minimize_direct` is more debatable and I think we can design the rule differently.

**Comment by ALeczkowski** [comment]
Thanks for the reply!

However, this is not exactly what I meant by my question.

If I understand correctly, the label `conj:emb` is non-existent until you introduce it in the rule `add_conj_emb`. When you look at the strat main function:

```
strat main {
  Seq (
    Onf (eud_to_ud),
    specific_expr_init, % Initialization of a specific treatment for MWE, titles and translated words.
    Onf (sub_relations), % Package SUB_RELATIONS must precede package REL_EXTENSIONS because this package swat extensions on level 2.
    **Onf (rel_extensions),**
    Onf (relations), % Package RELATIONS must precede REVERSE_RELATIONS.MAIN because the renaming of some relations depends on the cooccurrence with other relations (nsubj, csubjin the impersonal diathesis for instance)
    reverse_relations.main, % Reversing aux, case, cop and mark relations.
    Onf (minimize_conj_dep), % Package MINIMIZE_CONJ_DEP must follow strategy REVERSE_RELATIONS.MAIN because the shared dependents can depend on relation inversions.Moreover, a dependance can shift from a coordination to an ascendant. Ex: fr-ud-dev_00446 -entre 3 et 6 généralement
    **Onf (add_conj_emb)**, % Package ADD_CONJ_EMB must follow REVERSE_RELATIONS.MAIN because it depends on head changes ; Ex: UD_French-GSD fr-ud-dev_00751
    Onf (chained_relations),% Package CHAINED_RELATIONS must follow package SHIFT_RULES because it can entail a wrong coordination between two elements because of errors in head changes (UD_French-GS: fr-ud-dev_00131).
    specific_expr_close, % Closing a specific treatment for MWE, titles and translated words.
    Onf (unk_rel),
  )
}
```

the rule `add_conj_emb` is called after, for instance, the rule `change_ext` (which is the part of the package `rel_extensions`). As a result, at the moment of calling the rule `change_ext` there is no such label in the treebanks as `conj:emb`.

**Comment by perrier54** [comment]
You are right. The mention of `conj:emb` is useless in the rules `change_ext`, `minimize_direct`, `minimize_indirect_mod`, `minimize_indirect_conj-obj`. 

Since we have changed the order of rule execution many times, this may be a leftover from an old rule execution choice.
Currently I am not working on this conversion system, but as soon as I have time I will fix the rules.

Anyway, thanks for the comment !
